### PR TITLE
Handle response errors

### PR DESCRIPTION
**Pull request by Iyadhfaleh**
I think json_decode is better to handle responses APIs . 
"toArray" method will throw exception when status code >= 300 .

### CODE REVIEW

File: src/Bridge/Anthropic/ModelHandler.php [hunk 1]
```diff
@@ -55,13 +60,27 @@ public function request(Model $model, object|array|string $input, array $options
...
+     * @throws RedirectionExceptionInterface
+     * @throws ClientExceptionInterface
+     */
     public function convert(ResponseInterface $response, array $options = []): LlmResponse
     {
         if ($options['stream'] ?? false) {
             return new StreamResponse($this->convertStream($response));
         }
 
-        $data = $response->toArray();
```
Comment (OskarStark) [inline src/Bridge/Anthropic/ModelHandler.php:64]: What about `toArray(throw: false)` ?

Comment (Iyadhfaleh) [reply src/Bridge/Anthropic/ModelHandler.php:64]: You are right , done

**Comment by chr-hertel** [comment]
So, first of all thanks and I do agree that error handling is currently not in great shape.

the code that you added is mostly sitting on top of the expected json/array structure - which is a valid addition, but i doesn't help with 4xx/5xx situations, since everything is converted to a runtime exception and context of the original error or unexpected behavior would get lost.

i consider the structural changes that you check and error handling with RuntimeException a valuable addition, but would still keep the `toArray()` with throwing exceptions. nevertheless, this should be improved but rather in a different layer. and without loosing the original exception information.

this was the original handling, which is broken since i introduced the `AsyncResponse` i guess...
https://github.com/php-llm/llm-chain/blob/main/src/Platform.php#L44-L54

File: src/Bridge/Anthropic/ModelHandler.php [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 namespace PhpLlm\LlmChain\Bridge\Anthropic;
 
+use http\Exception\RuntimeException;
```
Comment (chr-hertel) [inline src/Bridge/Anthropic/ModelHandler.php:7]: use `PhpLlm\LlmChain\Exception\RuntimeException` instead

File: src/Bridge/Anthropic/ModelHandler.php [hunk 3]
```diff
@@ -55,13 +61,28 @@ public function request(Model $model, object|array|string $input, array $options
         ]);
     }
 
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws DecodingExceptionInterface
+     * @throws ClientExceptionInterface
+     */
```
Comment (chr-hertel) [inline src/Bridge/Anthropic/ModelHandler.php:70]: please remove this again. this is only needed due to PhpStorm inspections, not because of adding value. if you want to introduce valuable error information

File: src/Bridge/Anthropic/ModelHandler.php [hunk 4]
```diff
@@ -55,13 +61,28 @@ public function request(Model $model, object|array|string $input, array $options
...
     public function convert(ResponseInterface $response, array $options = []): LlmResponse
     {
         if ($options['stream'] ?? false) {
             return new StreamResponse($this->convertStream($response));
         }
 
-        $data = $response->toArray();
+        $data = $response->toArray(false);
```
Comment (chr-hertel) [inline src/Bridge/Anthropic/ModelHandler.php:77]: let's revert this change but keep the structural checks below

```suggestion
        $data = $response->toArray();
```

File: src/Bridge/Anthropic/ModelHandler.php [hunk 5]
```diff
@@ -55,13 +61,28 @@ public function request(Model $model, object|array|string $input, array $options
...
+
+        if (!isset($data['content']) || !count($data['content'])) {
```
Comment (chr-hertel) [inline src/Bridge/Anthropic/ModelHandler.php:79]: not a big fan of implicit casts, let's be explicit
```suggestion
        if (!isset($data['content']) || 0 === count($data['content'])) {
```

File: src/Bridge/Ollama/LlamaModelHandler.php [hunk 6]
```diff
@@ -41,7 +42,15 @@ public function request(Model $model, object|array|string $input, array $options
 
     public function convert(ResponseInterface $response, array $options = []): LlmResponse
     {
-        $data = $response->toArray();
+        $data = $response->toArray(false);
```
Comment (chr-hertel) [inline src/Bridge/Ollama/LlamaModelHandler.php:45]: same here, this code silences http error that are not handled below.

```suggestion
        $data = $response->toArray();
```

File: src/Bridge/OpenAI/Embeddings/ResponseConverter.php [hunk 7]
```diff
@@ -20,7 +21,11 @@ public function supports(Model $model, array|string|object $input): bool
 
     public function convert(ResponseInterface $response, array $options = []): VectorResponse
     {
-        $data = $response->toArray();
+        $data = $response->toArray(false);
```
Comment (chr-hertel) [inline src/Bridge/OpenAI/Embeddings/ResponseConverter.php:24]:
```suggestion
        $data = $response->toArray();
```

File: src/Bridge/OpenAI/GPT/ResponseConverter.php [hunk 8]
```diff
@@ -33,7 +33,7 @@ public function convert(HttpResponse $response, array $options = []): LlmRespons
             return $this->convertStream($response);
         }
 
-        $data = $response->toArray();
+        $data = $response->toArray(false);
```
Comment (chr-hertel) [inline src/Bridge/OpenAI/GPT/ResponseConverter.php:36]:
```suggestion
        $data = $response->toArray();
```

File: src/Bridge/Replicate/LlamaResponseConverter.php [hunk 9]
```diff
@@ -21,7 +22,11 @@ public function supports(Model $model, object|array|string $input): bool
 
     public function convert(HttpResponse $response, array $options = []): LlmResponse
     {
-        $data = $response->toArray();
+        $data = $response->toArray(false);
```
Comment (chr-hertel) [inline src/Bridge/Replicate/LlamaResponseConverter.php:25]:
```suggestion
        $data = $response->toArray();
```

File: src/Bridge/Voyage/ModelHandler.php [hunk 10]
```diff
@@ -39,7 +40,11 @@ public function request(Model $model, object|string|array $input, array $options
 
     public function convert(ResponseInterface $response, array $options = []): LlmResponse
     {
-        $response = $response->toArray();
+        $response = $response->toArray(false);
```
Comment (chr-hertel) [inline src/Bridge/Voyage/ModelHandler.php:43]:
```suggestion
        $response = $response->toArray();
```

File: src/Bridge/Voyage/ModelHandler.php [hunk 11]
```diff
@@ -4,6 +4,7 @@
 
 namespace PhpLlm\LlmChain\Bridge\Voyage;
 
+use http\Exception\RuntimeException;
```
Comment (chr-hertel) [inline src/Bridge/Voyage/ModelHandler.php:7]: let's use `PhpLlm\LlmChain\Exception\RuntimeException` instead

File: src/Bridge/Replicate/LlamaResponseConverter.php [hunk 12]
```diff
@@ -4,6 +4,7 @@
 
 namespace PhpLlm\LlmChain\Bridge\Replicate;
 
+use http\Exception\RuntimeException;
```
Comment (chr-hertel) [inline src/Bridge/Replicate/LlamaResponseConverter.php:7]: also `PhpLlm\LlmChain\Exception\RuntimeException`

File: src/Bridge/Ollama/LlamaModelHandler.php [hunk 13]
```diff
@@ -4,6 +4,7 @@
 
 namespace PhpLlm\LlmChain\Bridge\Ollama;
 
+use http\Exception\RuntimeException;
```
Comment (chr-hertel) [inline src/Bridge/Ollama/LlamaModelHandler.php:7]: `PhpLlm\LlmChain\Exception\RuntimeException` :)

**Comment by Iyadhfaleh** [comment]
Thank you for your review and your feedbacks @chr-hertel. I applied your suggestions

**Review by chr-hertel — APPROVED** [review]
Thank you @Iyadhfaleh
